natlas-server/app/scope/scope.py

```diff
@@ -104,39 +106,24 @@ def update(self):
 		self.updateScanManager()
 		log("ScopeManager Updated")
 
-	def address_in_collection(self, targetAddr, networkCollection):
-		""" Take in a collection of networks, identify if the target address is in one of those networks """
-		inCollection = False
-		for network in networkCollection:
-			# TODO this eventually needs to be upgraded to support IPv6
-			if str(network).endswith('/32') and targetAddr == ipaddress.IPv4Address(str(network).split('/32')[0]):
-				inCollection = True
-			if targetAddr in network:
-				inCollection = True
-		return inCollection
-
 	def isAcceptableTarget(self, target):
 		# Ensure it's a valid IPv4Address
 		try:
 			# TODO this eventually needs to be upgraded to support IPv6
-			targetAddr = ipaddress.IPv4Address(target)
-		except ipaddress.AddressValueError:
+			targetAddr = IPAddress(target)
+		except AddrFormatError:
 			return False
 
 		# if zero, update to make sure that the scopemanager has been populated
 		if self.getScopeSize() == 0:
 			self.update()
 
-		inScope = self.address_in_collection(targetAddr, self.getScope())
-
 		# Address doesn't fall in scope ranges
-		if not inScope:
+		if not self.scanmanager.inWhitelist(target):
```

Comment:
Let's consolidate this line with the line below. It'll reduce the number of code paths in the function and can easily be explained by changing the above comment to "Address doesn't fall in scope ranges or is blacklisted"